src/components/reusables/Filter/OptionDisplay.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import { Checkbox } from '@digdir/design-system-react';
+import { SearchField } from '@altinn/altinn-design-system';
+import { useTranslation } from 'react-i18next';
+
+import { arraysEqual } from '@/resources/utils';
+import { usePrevious } from '@/resources/hooks';
+
+import { optionSearch } from './utils';
+import type { FilterOption } from './utils';
+import classes from './OptionDisplay.module.css';
+
+export interface OptionDisplayProps {
+  options: FilterOption[];
+  value?: string[];
+  onValueChange?: (value: string[]) => void;
+  searchable?: boolean;
+  compact?: boolean;
+}
+
+export const OptionDisplay = ({
+  options,
+  value,
+  searchable,
+  compact = true,
+  onValueChange,
+}: OptionDisplayProps) => {
+  const [selectedValues, setSelectedValues] = useState<string[]>(value ?? []);
+  const [sortedOptions, setSortedOptions] = useState<FilterOption[]>(options);
+  const { t } = useTranslation('common');
+
+  // Update selected values when there are external changes
+  const prevValue = usePrevious(value);
+  useEffect(() => {
+    if (!arraysEqual(value, prevValue)) {
+      setSelectedValues(value ?? []);
+    }
+  }, [value]);
+
+  // Update sorted options when there are external changes
+  const prevOptions = usePrevious(options);
+  useEffect(() => {
+    if (!arraysEqual(options, prevOptions)) {
+      setSortedOptions(options);
+    }
+  }, [options]);
+
+  const changeHandler = (newValues: string[], addedValue?: string) => {
+    setSelectedValues(newValues);
+    onValueChange?.(newValues);
+  };
+
+  const handleSelection = (selectedValue: string) => {
+    if (selectedValues?.includes(selectedValue)) {
+      changeHandler(selectedValues.filter((value) => value !== selectedValue));
+    } else {
+      changeHandler([...selectedValues, selectedValue], selectedValue);
+    }
+  };
+
+  const handleSearch = (searchString: string) => {
+    setSortedOptions(optionSearch(options, searchString));
+  };
+
+  const checkboxes = sortedOptions.map((option, index) => {
+    const isSelected = selectedValues?.includes(option.value);
+    return (
+      <button
+        className={classes.option}
+        key={option.value}
+        onClick={() => {
+          handleSelection(option.value);
+        }}
+        tabIndex={-1}
+      >
+        {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
+        <div
+          className={classes.propagationStopper}
+          tabIndex={-1}
+          onClick={(event) => {
+            event.stopPropagation();
+          }}
+          onKeyDown={(event) => {
+            if (event.key !== 'Escape') {
+              event.stopPropagation();
+            }
+          }}
+        >
+          <Checkbox
```

Comment:
The cous state looks a bit buggy on the first element:
<img width="292" alt="image" src="https://github.com/Altinn/altinn-access-management-frontend/assets/31471142/153cac98-2db6-4f2d-baf6-c385f257ef87">